core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -964,8 +964,7 @@ private ExecutionModel parseCreateMatView(
                 }
             }
 
-            final String matViewSql = Chars.toString(lexer.getContent(), startOfQuery, endOfQuery);
-            tableOpBuilder.setSelectText(matViewSql, startOfQuery);
+            tableOpBuilder.setSelectText(lexer.getContent().subSequence(startOfQuery, endOfQuery), startOfQuery);
```

Comment:
this is a lot less explicit that toString and thus will be error prone. tableOp is a detachable object, it cannot depend on mutable objects. subSequence() returns potentially mutable interface

Reply:
changed this back in https://github.com/questdb/questdb/pull/5621/commits/a57460aa96808e6e7fa7d918f9971f70565cbbd1